x-pack/plugins/search_playground/public/components/search_mode/result_list.tsx

```diff
@@ -5,54 +5,114 @@
  * 2.0.
  */
 
+import React, { useEffect, useState } from 'react';
+
 import {
   EuiFlexGroup,
   EuiFlexItem,
   EuiHorizontalRule,
+  EuiPagination,
   EuiPanel,
   EuiText,
   EuiTitle,
 } from '@elastic/eui';
-import React from 'react';
 
-const DEMO_DATA = [
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-  { id: '123321', name: 'John Doe', age: 25 },
-];
+import { UnifiedDocViewerFlyout } from '@kbn/unified-doc-viewer-plugin/public';
+
+import type { DataView } from '@kbn/data-views-plugin/public';
+import type { EsHitRecord } from '@kbn/discover-utils/types';
+import type { SearchHit } from '@elastic/elasticsearch/lib/api/types';
+import { buildDataTableRecord } from '@kbn/discover-utils';
+import { i18n } from '@kbn/i18n';
+import { Pagination } from '../../types';
+import { getPageCounts } from '../../utils/pagination_helper';
+import { EmptyResults } from './empty_results';
+import { useKibana } from '../../hooks/use_kibana';
+
+export interface ResultListArgs {
+  searchResults: SearchHit[];
+  pagination: Pagination;
+  onPaginationChange: (nextPage: number) => void;
+}
 
-export const ResultList: React.FC = () => {
+export const ResultList: React.FC<ResultListArgs> = ({
+  searchResults,
+  pagination,
+  onPaginationChange,
+}) => {
+  const {
+    services: { data },
+  } = useKibana();
+  const [dataView, setDataView] = useState<DataView | null>(null);
+  useEffect(() => {
+    data.dataViews.getDefaultDataView().then((d) => setDataView(d));
+  }, [data]);
+  const [flyoutDocId, setFlyoutDocId] = useState<string | undefined>(undefined);
+  const { totalPage, page } = getPageCounts(pagination);
+  const hit =
+    flyoutDocId &&
+    buildDataTableRecord(searchResults.find((item) => item._id === flyoutDocId) as EsHitRecord);
   return (
     <EuiPanel grow={false}>
       <EuiFlexGroup direction="column" gutterSize="none">
-        {DEMO_DATA.map((item, index) => {
+        {searchResults.map((item, index) => {
```

Comment:
suggestion: use the `packages/kbn-search-index-documents` for these result components.